Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrExtensions.kt

```diff
@@ -86,4 +86,15 @@ fun ConstantStub.determineConstantAnnotationClassifier(): Classifier = when (thi
         8 -> "Double"
         else -> error("Real constant with unexpected size of $size.")
     }
-}.let { Classifier.topLevel(cinteropInternalPackage, "ConstantValue").nested(it) }
\ No newline at end of file
+}.let { Classifier.topLevel(cinteropInternalPackage, "ConstantValue").nested(it) }
+
+/**
+ * Returns the original name of the given type.
+ */
+val StubType.underlyingTypeFqName: String
```

Comment:
Should it be the abstract property of StubType instead of extension? Dynamic type check `when(this) is ...` smells like an antipattern, isn't it?

Reply:
It is easy to replace extension with abstract method if we see that this method affects performance. Until then I prefer to keep classes as simple as possible.